### PR TITLE
R project sprint

### PR DESCRIPTION
**Pull request by shannonpileggi**

### CODE REVIEW

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+enhancement of this code base is upheld by volunteers. Those who make the 
+executive decisions are the [R Core Team](https://www.r-project.org/contributors.html){target="_blank"}; however, 
+community members are still encouraged to contribute.
+
+The aim of the R Project Sprint as I perceive it was to foster mentorship, collaborations,
+and personal relationships between the novice and experienced contributors, reducing
+the barrier to one's first contribution.
+
+The purpose of this post is describe my personal experience at this event; a comprehensive 
+guide for how contribute to base R is documented in the [R Contribution Working Group website](https://contributor.r-project.org/){target="_blank"}.
```
Comment (hfrick) [inline _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:59]:
```suggestion
guide for how contribute to base R is documented on the [R Contribution Working Group website](https://contributor.r-project.org/){target="_blank"}.
```

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+translation).
+
+* Read, modify, or write R source code (bugs and enhancements).
+
+* Read, modify, or write C source code (bugs and enhancements).
+
+# Workflows for contributions
+
+
+During the sprint, the progress on these items was tracked [https://github.com/r-devel/r-project-sprint-2023/issues](https://github.com/r-devel/r-project-sprint-2023/issues){target="_blank"}. Either the initial report or the final resolution of these items may be seen in the R Bug Tracking System, known as [Bugzilla](https://bugs.r-project.org/){target="_blank"}.
```
Comment (hfrick) [inline _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:121]: was tracked _at_ <URL>? or did you mean to turn a word into a link here?

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+Depending on the nature of the proposed change, the contributor may need
+to build R from source. The instructions regarding how to do so on your personal computing
+enviroment are outlined in the [R Patched and Development Versions Chapter](https://contributor.r-project.org/rdevguide/GetStart.html){target="_blank"} of the [R Development Guide](https://contributor.r-project.org/rdevguide/index.html){target="_blank"}. Alternatively, one could also build R from source without touching 
+their local computing environment using the [GitHub Code Space
+R Dev Containter](https://github.com/r-devel/r-dev-env){target="_blank"} (which I
```
Comment (hfrick) [inline _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:127]:
```suggestion
R Dev Container](https://github.com/r-devel/r-dev-env){target="_blank"} (which I
```

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+heard facilitated collaboration nicely as well).
+
+Regardless of whether you needed to build R from source, proposed changes could then be tested on multiple computing platforms via a pull request to [https://github.com/r-devel/r-svn](https://github.com/r-devel/r-svn){target="_blank"},
+which mirrors the official base R server. After tests have passed, the contributor could 
+extract the diff from the pull request to submit a proposed solution via Bugzilla.
+
+Once a proposal is submitted,  member(s) of the R core team review. If the 
+proposed change modifies the code base, additional checks are run against
+all packages on CRAN (which there are currently ~20,000) to determine
```
Comment (hfrick) [inline _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:136]: I think not all changes are checked against all of CRAN but rather a subset (which is still large).

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+Depending on the nature of the proposed change, the contributor may need
+to build R from source. The instructions regarding how to do so on your personal computing
+enviroment are outlined in the [R Patched and Development Versions Chapter](https://contributor.r-project.org/rdevguide/GetStart.html){target="_blank"} of the [R Development Guide](https://contributor.r-project.org/rdevguide/index.html){target="_blank"}. Alternatively, one could also build R from source without touching 
+their local computing environment using the [GitHub Code Space
+R Dev Containter](https://github.com/r-devel/r-dev-env){target="_blank"} (which I
+heard facilitated collaboration nicely as well).
+
+Regardless of whether you needed to build R from source, proposed changes could then be tested on multiple computing platforms via a pull request to [https://github.com/r-devel/r-svn](https://github.com/r-devel/r-svn){target="_blank"},
+which mirrors the official base R server. After tests have passed, the contributor could 
+extract the diff from the pull request to submit a proposed solution via Bugzilla.
```
Comment (hfrick) [inline _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:132]: This workflow is also described at https://contributor.r-project.org/rdevguide/FixBug.html#using-a-git-mirror

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+fixes and enhancements were implemented throughout the week. 
+
+Time zone differences and personal availability among the R Core Team and contributors 
+can lead to time lags in communications, losing momentum for initiated issues. Having
+contributors in person together facilitated live and immediate feedback, allowing
+for faster iteration and completion.
+
+I can now confidently either submit a new bug report or address an 
+existing bug report. Moreover, I understand workflows for contributions, resources for help,
+and how to interact with the community to for help should I get stuck.
```
Comment (hfrick) [inline _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:179]:
```suggestion
and how to interact with the community for help should I get stuck. 
```

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+The [R Project Sprint participants](https://contributor.r-project.org/r-project-sprint-2023/participants/){target="_blank"} were impressively diverse both in terms
+of professional experience and geographic residence. A non-exhaustive list of countries
+represented at the in-person event includes Argentina, Brazil, Canada, England, India, Hungary, Nepal, Netherlands, New Zealand, Nigeria, Oman, and Senegal, Switzerland, and United States.
+
+Some individuals participated remotely via video conference, either by choice
+or circumstance. The combination of [the United Kingdom's air traffic
+control issues](https://www.bbc.com/news/uk-66637156){target="_blank"} and [train
+strikes](https://www.bbc.com/news/business-66479059){target="_blank"} caused delayed
+arrival for some.
```
Comment (hfrick) [inline _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:91]: Come to the UK they said, it'll be great they said 🙈

**Review by hfrick — COMMENTED** [review]
Great post!

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd (hunk 2 above)
Comment (shannonpileggi) [reply _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:121]: yep, i meant at

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd (hunk 4 above)
Comment (shannonpileggi) [reply _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd:136]: k i have asked heather if she wants to review as well, maybe she can confirm

File: _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd (hunk 5 above)
Comment (shannonpileggi) [reply _posts/2023-09-07-r-project-sprint/r-project-sprint.Rmd]: thanks, link added in!

**Pull request merged**